### PR TITLE
Add flag to use Yarn

### PR DESCRIPTION
**Pull request by gowrizrh**
When a package runs install scripts globally the pipe fails

image

I've searched through multiple forums but only found a solution in some obscure PR comment and the only resolution mentioned by **Atlassian** is `unset NPM_CONFIG_USER`

https://github.com/expo/sentry-expo/pull/26#issuecomment-453822980

### CODE REVIEW

**Comment by gowrizrh** [comment]
This was a never ending deep rabbit hole.

TLDR: `node ^v16.11` comes with `npmcli ^v8.0` which has a dependency of `@npmcli/promise-spawn ^v1.32.0`. `promise-spawn` for some whatever elegant reason decides take hands on it's own and run npm scripts as nobody if given the user is root.

image

This was documented, then removed then re-added but the documentation was never readded

### Details

https://github.com/npm/promise-spawn/blob/v1.3.2/index.js#L14

This was removed in `^2.0` but `npmcli/exec` is not ready so was re-added and released as `^3.0`

https://github.com/npm/promise-spawn/pull/12#issue-1193638785

https://github.com/npm/promise-spawn/issues/6#issuecomment-1156810900

Node 16.15 bumps npmcli to `8.11` which still uses promise-spawn v3 anyway so updating our images to `node 16.15` won't fix this issue. This is an upstream issue in the node alpine community too.

https://github.com/nodejs/docker-node/issues/1734


The only workaround was ditch npm and use yarn. Our deploy pipe does not support running via yarn as of now which is exactly what this PR adds.

**Review by AdamJHall — APPROVED** [review]
Good explanation and references :+1:
